### PR TITLE
feat(fmt): Add `deno-fmt-ignore` and `deno-fmt-ignore-file` comment support

### PR DESCRIPTION
**Pull request by dsherret**
This PR changes the ignore comments to be `// deno-fmt-ignore` and `// deno-fmt-ignore-file`, which was implemented by making these configurable and then specified in the "deno" top level config.

See https://github.com/dprint/dprint/issues/196 for implementation details.

**`deno-fmt-ignore` Example**

Formats as-is:

```ts
// deno-fmt-ignore
const identity = [
    1, 0, 0,
    0, 1, 0,
    0, 0, 1,
];

const identity = /* deno-fmt-ignore */ [
    1, 0, 0,
    0, 1, 0,
    0, 0, 1,
];
```

**`deno-fmt-ignore-file` Example**

Formats as-is:

```ts
// deno-fmt-ignore-file
some

.random.code(       5 );
call(   1   ,    5  );
```

Currently that must be the first comment in the file, but in the future I'll improve that to just have to be in a comment before the first node.

### CODE REVIEW

File: cli/fmt.rs [hunk 1]
```diff
@@ -21,21 +21,38 @@ use std::path::PathBuf;
 use std::sync::atomic::{AtomicUsize, Ordering};
 use std::sync::{Arc, Mutex};
 
-fn is_supported(path: &Path) -> bool {
```
Comment (dsherret) [inline cli/fmt.rs:24]: I just moved some functions around in here so this reads top to bottom, which I think helps improve the readability of this file a little more.

File: cli/Cargo.toml [hunk 2]
```diff
@@ -33,7 +33,7 @@ byteorder = "1.3.4"
 clap = "2.33.0"
 dirs = "2.0.2"
 dlopen = "0.1.8"
-dprint-plugin-typescript = "0.14.1"
+dprint-plugin-typescript = "0.16.0"
```
Comment (dsherret) [inline cli/Cargo.toml:36]: Question: How do we tell people about `deno-fmt-ignore` and `deno-fmt-ignore-file`?

Comment (ry) [reply cli/Cargo.toml:36]: There are two places we should do this: 

1. We should add a manual section for "deno fmt" in std/manual.md
2. I think adding it to the cli help messages would be helpful too https://github.com/denoland/deno/blob/796fc9bc3e8e4da7d53fb4617511ce4e2be22485/cli/flags.rs#L597-L604

**Comment by SyrupThinker** [comment]
Wouldn't `@deno-fmt-ignore` and `@deno-fmt-ignore-file` more consistent with the rest of the JS/TS ecosystem for processing directives?

**Comment by dsherret** [comment]
@SyrupThinker this was done because it's the format that prettier does (see [here](https://prettier.io/docs/en/ignore.html#javascript)). Other tools like eslint and tslint also use similar comments.

I'm not sure using an `@` symbol would be more consistent with the rest of the JS/TS ecosystem, but it would definitely be more consistent with the TS compiler's `@ts-ignore` and `@ts-expect` comments. Considering Deno supports these comments via the TS compiler then perhaps we should copy the ts compiler and use `@` as you suggested? Or perhaps support an optional `@` symbol at the front, but only document using an `@` symbol?

File: cli/Cargo.toml (hunk 2 above)
Comment (dsherret) [reply cli/Cargo.toml:36]: I updated the docs.

Note: there is a pending question here that should be considered https://github.com/denoland/deno/pull/5075#issuecomment-623594243

File: std/manual.md [hunk 3]
```diff
@@ -1298,6 +1298,40 @@ All listeners added using `window.addEventListener` were run, but
...
+deno fmt myfile1.ts myfile2.ts
+# check if all the JS/TS files in the current directory and subdirectories are formatted
+deno fmt --check
+# format stdin and write to stdout
+cat file.ts | deno fmt -
+```
+
+Ignore code to format by preceding it with a `// deno-fmt-ignore` comment:
+
+<!-- prettier-ignore-start -->
```
Comment (dsherret) [inline std/manual.md:1319]: One day I will finish the markdown formatter... (I at least have the JSON formatter in Rust now, but still a bit of work left on the markdown one).

**Comment by nayeemrmn** [comment]
> Wouldn't `@deno-fmt-ignore` and `@deno-fmt-ignore-file` be more consistent with the rest of the JS/TS ecosystem for processing directives?

-1, less symbols is less opinionated. Maybe if there was a good reason but other tools have been fine without it.

**Comment by ry** [comment]
I agree with @nayeemrmn

**Pull request merged**